sdk/avalon_sdk/work_order/work_order_params.py

```diff
@@ -126,8 +132,11 @@ def set_session_key_iv(self, session_iv):
 
     def set_requester_nonce(self, requester_nonce):
         """Set requesterNonce work order parameter."""
-        if not requester_nonce.strip():
-            raise TypeError("requesterNonce cannot be None")
+        if requester_nonce is None or \
+                not is_valid_hex_str(requester_nonce) or \
+                len(requester_nonce) != 32:
+            raise TypeError("Empty or Invalid data \
```

Comment:
Let this also be a ValueError. Can check for TypeError explicitly.

Reply:
Done